frame/contracts-registry/src/lib.rs

```diff
@@ -0,0 +1,317 @@
+		/// Add judge
+		///
+		/// The dispatch origin of this call must be _Signed_ and the sender must
+		/// have funds to cover the deposit.
+		///
+		/// - `judge`: Account that will be added as one of a judges
+		///
+		/// Emits `AddedJudge`.
+		#[pallet::call_index(0)]
+		#[pallet::weight(T::WeightInfo::add_judge())]
+		pub fn add_judge(origin: OriginFor<T>, judge: T::AccountId) -> DispatchResult {
+			let _ = T::AddJudgeOrigin::ensure_origin(origin)?;
+
+			let real_judges_count = Judges::<T, I>::decode_len().unwrap_or(0) as u32;
+			Judges::<T, I>::try_append(&judge).map_err(|_| Error::<T, I>::TooMany)?;
+
+			Self::deposit_event(Event::AddedJudge {
+				judge_id: real_judges_count + 1,
+				deposit: judge,
+			});
+			Ok(())
+		}
+
+		/// Judge can sign contract
+		///
+		/// The dispatch origin of this call must be _Signed_ and the sender must
+		/// have funds to cover the deposit.
+		///
+		/// - `contract_id`: The `id` of a contract that should be signed
+		///
+		/// Emits `JudgeSigned`.
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::judge_sign_contract())]
+		pub fn judge_sign_contract(
+			origin: OriginFor<T>,
+			contract_id: ContractIndex,
+		) -> DispatchResult {
+			let who = T::SubmitOrigin::ensure_origin(origin)?;
+			let is_judge = Judges::<T, I>::get().contains(&who);
+			ensure!(is_judge, Error::<T, I>::NotJudge);
+
+			JudgeSignatures::<T, I>::try_mutate(contract_id, |signatures_list: &mut Option<BoundedVec<T::AccountId, T::MaxSignatures>>| -> DispatchResult {
+				let signatures = signatures_list.as_mut().ok_or(Error::<T, I>::ContractNotFound)?;
+				let already_signed = signatures.contains(&who);
+				ensure!(!already_signed, Error::<T, I>::AlreadySigned);
+
+				signatures.try_push(who.clone()).map_err(|_| Error::<T, I>::TooMany)?;
+				Ok(())
+			})?;
+
+			Self::deposit_event(Event::JudgeSigned { contract_id, signer: who });
+			Ok(())
+		}
+
+		/// Any caller can create contract
+		///
+		/// The dispatch origin of this call must be _Signed_ and the sender must
+		/// have funds to cover the deposit.
+		///
+		/// - `data`: A content of contract
+		///
+		/// Emits `ContractCreated`.
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::create_contract(T::MaxContractContentLen::get()))]
+		pub fn create_contract(
+			origin: OriginFor<T>,
+			data: BoundedVec<u8, T::MaxContractContentLen>,
+			parties: BoundedVec<T::AccountId, T::MaxSignatures>,
+		) -> DispatchResult {
+			let who = T::SubmitOrigin::ensure_origin(origin)?;
+
+			let required_deposit = Self::calculate_deposit(&data);
+			T::Currency::reserve_named(T::ReserveIdentifier::get(), &who, required_deposit)?;
+
+			let index = Self::contracts_count();
+			Contracts::<T, I>::insert(
+				index,
+				ContractDataStorage::<T::MaxContractContentLen, T::MaxSignatures, T::AccountId> {
+					data,
+					parties,
+					creator: who.clone(),
+				},
+			);
+
+			JudgeSignatures::<T, I>::insert(index, BoundedVec::default());
```

Comment:
```suggestion
			JudgesSignatures::<T, I>::insert(index, BoundedVec::default());
```
For consistency with `Parties` below.